subprojects/core/src/main/java/org/gradle/api/internal/GradleInternal.java

```diff
@@ -43,7 +43,7 @@
  * consumption.
  */
 @UsedByScanPlugin
-@ServiceScope(Scope.Build.class)
+@ServiceScope({Scope.Build.class, Scope.Settings.class})
```

Comment:
💭 Just mentioning here again that these multi-scoped things are a code smell suggesting that service scope should not be declared on the service interfaces.